lib/common/widgets/map_bottom_sheet.dart

```diff
@@ -40,14 +40,15 @@ class _MapBottomSheetState extends ConsumerState<MapBottomSheet> {
   Widget build(BuildContext context) {
     final mode = ref.watch(sheetModeProvider);
     final trigger = ref.watch(sheetTriggerProvider);
+
     final sheetPosition =
         mode == SheetMode.half ? BottomSheetConfig.halfSizePercent : BottomSheetConfig.fullSizePercent;
 
     if (trigger) {
       Future.microtask(() async {
         ref.read(sheetTriggerProvider.notifier).state = false;
         await controller.animateTo(
-          BottomSheetConfig.hiddenSizePercent,
+          ref.read(sheetStateProvider) == SheetState.hidden ? BottomSheetConfig.hiddenSizePercent : sheetPosition,
```

Comment:
because of this change, in the route details screen, after pressing the end route button, you should not only set the bottomSheetTrigger to true, but also set the bottomSheetState to hidden. Because now pressing the button does not cause the sheet to hide. And because of this, if someone ends the route and then starts a new one, the icons on the map will be grayed out at the beginning, because their color is decided basing on the sheet state (hidden or not).